kloppy/domain/models/tracking.py

```diff
@@ -1,26 +1,45 @@
 from dataclasses import dataclass
 from typing import List, Dict
 
-from .common import Dataset, DataRecord
+from .common import Dataset, DataRecord, Ground
 from .pitch import Point
 
 
 @dataclass
 class Frame(DataRecord):
     frame_id: int
-    home_team_player_positions: Dict[str, Point]
-    away_team_player_positions: Dict[str, Point]
+    players_coordinates: Dict[str, Point]
+    players_ground: Dict[str, Ground]
     ball_position: Point
 
+    @property
+    def home_team_players_coordinates(self):
+        return {
+            player_id: coordinates
+            for (player_id, coordinates) in self.players_coordinates.items()
+            if self.players_ground[player_id] == Ground.HOME
+        }
+
+    @property
+    def away_team_players_coordinates(self):
+        return {
+            player_id: coordinates
+            for (player_id, coordinates) in self.players_coordinates.items()
+            if self.players_ground[player_id] == Ground.AWAY
+        }
+
```

Comment:
To be able to set [home, away]_team_players_coordinates as a property, I had to add a players_ground property to the Frame class, because from the Frame class I didn't have access to meta_data.teams.players. 

It works, but I'm not sure if there is a better way to do this.

Reply:
I checked the dataclass and I think we should use the Player object itself as a key. The Player dataclass needs to be frozen.

Then you can do:
```
@property
def away_team_players_coordinates(self):
   return {player.id: coordinates for player, coordinates in self.player_coordinates.items() if player.team.ground == Ground.AWAY}
```

Reply:
@koenvo ok! I tried to implement it but run with a potential conflict. In order to be able to use Player as a key, Team also needs to be frozen. However that might not be ideal as you might want to initialize a team without players and fill in the players later on? What do you think?

Reply:
Interesting problem. This circular reference makes a bit harder. The `frozen=True` is not required for the hash to work, you can also use `unsafe_hash=True`. In the meanwhile I'll check if there is a clean way to do it using `frozen`.

Reply:
Tried using `unsafe_hash=True` for Team while it works for the previous error now I get another error, this time `TypeError: unhashable type: 'list'`. From what I was able to understand, the problem here is that `players` property of Teams is a list of players and list can't be used as key (tuples can, but not lists). The solution would be to make `Team.players` a tuple, but I think that also can bring more problems than solutions, as there are probably cases in which you need to add or remove players, to a Team after initializing it, don't you think?

Another option may be could be to pass a reference of the metadata to each frame object? From what I understand it would be a reference and not a copy (as when doing `player.team = Team`) so that shouldn't add any memory / performance issues?

Reply:
Didn't look at the lists.. sorry about that. Could have saved you some time there..
I think it's easier to fix this is create a custom hash function on team:
```
# see https://docs.python.org/3/reference/datamodel.html#object.__hash__

def __hash__(self):
   return hash(self.team_id)

def __eq__(self, other):
   if not isinstance(other, Team):
      return False
    return self.team_id == other.team_id
```

This way all other properties are ignored, which makes sense in this case.

Shouldn't we just get rid of the `away_team_players_coordinates` and `home_team_players_coordinates` properties? And make sure we have `Player` objects as keys. That would make it very easy for end-users to filter on home/away team - and other properties.

Reply:
Alright! Have a look at the updated code, comments about what needed to be done to make Player as a key worked are there. One thing that may be is relevant, is that  using player as a key increased the time to `load_epts_tracking_data` by ~50%. From 20s to 30s con my computer.

Reply:
Cool that’s working!

When performance is really an issue we can optimize it. Have already some ideas. So no worries for now